src/adapter/src/catalog.rs

```diff
@@ -392,10 +396,19 @@ impl CatalogState {
             .unwrap_or_else(|| panic!("unknown cluster {cluster_id}"))
     }
 
+    fn get_cluster_mut(&mut self, cluster_id: ClusterId) -> &mut Cluster {
+        self.try_get_cluster_mut(cluster_id)
+            .unwrap_or_else(|| panic!("unknown cluster {cluster_id}"))
```

Comment:
I'll give this a shot, but if I remember correctly clippy will complain because `.expect(...)` will not lazily evaluate the string while `.unwrap_or_else(...)` will.